### PR TITLE
Support `normalization_resolver` in `MLP`

### PR DESCRIPTION
**Pull request by lightaime**
This pull request added support for the `normalization_resolver` in `MLP`.

### CODE REVIEW

File: torch_geometric/nn/models/mlp.py [hunk 1]
```diff
@@ -76,18 +79,25 @@ def __init__(
...
     ):
         super().__init__()
 
-        act_first = act_first or relu_first  # Backward compatibility.
-        batch_norm_kwargs = batch_norm_kwargs or {}
+        # Backward compatibility:
+        act_first = act_first or kwargs.get("relu_first", False)
+        batch_norm = kwargs.get("batch_norm", None)
+        if batch_norm is not None and isinstance(batch_norm, bool):
+            warnings.warn("Argument `batch_norm` will be deprecated, "
```
Comment (Padarn) [inline torch_geometric/nn/models/mlp.py:96]: I'm wondering if we should decide on a version in which it will be depreciated and note that here? WDYT?

File: torch_geometric/nn/models/mlp.py [hunk 2]
```diff
@@ -114,11 +124,15 @@ def __init__(
...
+            if norm is not None:
+                norm_layer = normalization_resolver(
+                    norm,
+                    hidden_channels,
+                    **(norm_kwargs or {}),
+                )
             else:
-                norm = Identity()
-            self.norms.append(norm)
+                norm_layer = Identity()
```
Comment (Padarn) [inline torch_geometric/nn/models/mlp.py:134]: do you need to do this? could we just have no norm layer in the case that none are needed rather than adding the identity?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## [2.0.5] - 2022-MM-DD
 ### Added
+- Added Support for `normalization_resolver` in `MLP` ([#4951](https://github.com/pyg-team/pytorch_geometric/pull/4951))
```
Comment (Padarn) [inline CHANGELOG.md:8]: maybe we should add this to line 10 instead? the changelog is getting very big 😓

File: torch_geometric/nn/models/mlp.py [hunk 4]
```diff
@@ -48,23 +52,22 @@ class MLP(torch.nn.Module):
...
+            respective normalization function defined by :obj:`norm`.
             (default: :obj:`None`)
         plain_last (bool, optional): If set to :obj:`False`, will apply
             non-linearity, batch normalization and dropout to the last layer as
             well. (default: :obj:`True`)
         bias (bool, optional): If set to :obj:`False`, the module will not
             learn additive biases. (default: :obj:`True`)
-        relu_first (bool, optional): Deprecated in favor of :obj:`act_first`.
-            (default: :obj:`False`)
+        **kwargs (optional): Additional arguments of the MLP layer.
```
Comment (rusty1s) [inline torch_geometric/nn/models/mlp.py:70]:
```suggestion
        **kwargs (optional): Additional deprecated arguments of the MLP layer.
```

File: torch_geometric/nn/models/mlp.py (hunk 1 above)
Comment (rusty1s) [inline torch_geometric/nn/models/mlp.py:96]:
```suggestion
            warnings.warn("Argument `batch_norm` is deprecated, "
```

File: torch_geometric/nn/models/mlp.py (hunk 2 above)
Comment (rusty1s) [reply torch_geometric/nn/models/mlp.py:134]: This shifts the efforts of checking this to the `forward` call. I think either way is fine.

**Review by rusty1s — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (lightaime) [reply CHANGELOG.md:8]: Good idea!

File: torch_geometric/nn/models/mlp.py (hunk 1 above)
Comment (lightaime) [reply torch_geometric/nn/models/mlp.py]: I am not sure when we want to fully deprecate it though. @rusty1s any idea on this?

**Comment by lightaime** [comment]
- [ ] TODO: fix `norm` support in examples. Will make a seperate for it.

**Review by Padarn — APPROVED** [review]

**Comment by rusty1s** [comment]
Thank you @lightaime. Looking forward to the follow-up PR.
